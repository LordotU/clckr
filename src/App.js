/**
 * @format
 * @flow
 */

import React, { useState, useEffect } from 'react';
import {
  SafeAreaView,
  StatusBar,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import useStyles from './AppStyles';

export default function(): React$Node {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  const [count, setCount] = useState(0);
  const [highContrast, setHighContrast] = useState(false);

  const incrementCounter = () => setCount(count + 1);
  const resetCounter = () => setCount(0);
  const switchContrast = () => setHighContrast(!highContrast);

  const styles = useStyles(highContrast);

  return (
    <>
      <StatusBar barStyle={highContrast ? 'light-content' : 'dark-content'} />
      <SafeAreaView style={styles.outerView}>
        <TouchableWithoutFeedback
          accessible={true}
          accessibilityLabel="Increment counter"
          onPress={incrementCounter}>
          <View style={styles.innerView}>
            <Text style={styles.clickerCount} testID="counter">
              {count}
            </Text>
          </View>
        </TouchableWithoutFeedback>

        <TouchableOpacity
          accessible={true}
          accessibilityLabel="Reset counter"
          style={styles.button}
          onPress={resetCounter}>
          <Text style={styles.buttonInner}>Reset counter</Text>
        </TouchableOpacity>

        <TouchableOpacity
          accessible={true}
          accessibilityLabel="Switch contrast"
          style={styles.button}
          onPress={switchContrast}>
          <Text style={styles.buttonInner}>Switch contrast</Text>
        </TouchableOpacity>
      </SafeAreaView>
    </>
  );
}
