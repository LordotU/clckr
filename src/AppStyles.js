/**
 * @format
 * @flow
 */

import { StyleSheet } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

const FONT_SIZE_100 = 240;
const FONT_SIZE_100_DISPLAY_HEIGHT = 1242;

const defaults = {
  outerView: {
    flex: 1,
  },
  infoButton: {
    padding: 8,
    position: 'absolute',
    right: 24,
    top: 48,
    zIndex: 1,
  },
  infoIcon: {
    height: 32,
    width: 32,
  },
  innerView: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  clickerCount: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(FONT_SIZE_100, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
  button: {
    paddingHorizontal: 32,
    paddingVertical: 12,
    width: '100%',
  },
  buttonInner: {
    borderColor: 'black',
    borderWidth: 0.5,
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: 24,
    padding: 8,
    textAlign: 'center',
    width: '100%',
  },
};

const contrast = {
  ...defaults,
  outerView: {
    ...defaults.outerView,
    backgroundColor: 'black',
  },
  clickerCount: {
    ...defaults.clickerCount,
    color: 'white',
  },
  buttonInner: {
    ...defaults.buttonInner,
    backgroundColor: 'black',
    borderColor: 'white',
    color: 'white',
  },
};

const stylesDefaults = StyleSheet.create(defaults);

const stylesContrast = StyleSheet.create(contrast);

export default function(useContrast: boolean) {
  return useContrast ? stylesContrast : stylesDefaults;
}
