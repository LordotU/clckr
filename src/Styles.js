/**
 * @format
 * @flow
 */

import { StyleSheet } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

const FONT_SIZE_100_DISPLAY_HEIGHT = 1242;

const defaults = {
  outerView: {
    flex: 1,
  },
  infoButton: {
    padding: 8,
    position: 'absolute',
    right: 24,
    top: 40,
    zIndex: 1,
  },
  infoIcon: {
    height: 32,
    width: 32,
  },
  innerView: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  clickerCount: {
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(240, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
  button: {
    paddingHorizontal: 32,
    paddingVertical: 12,
    width: '100%',
  },
  buttonInner: {
    borderColor: 'black',
    borderWidth: 0.5,
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: 24,
    padding: 8,
    textAlign: 'center',
    width: '100%',
  },
  info: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  outerInfoLink: {
    padding: 8,
  },
  infoLink: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(48, FONT_SIZE_100_DISPLAY_HEIGHT),
    textDecorationLine: 'underline',
  },
  infoVersion: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(40, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
};

const contrast = {
  ...defaults,
  outerView: {
    ...defaults.outerView,
    backgroundColor: 'black',
  },
  clickerCount: {
    ...defaults.clickerCount,
    color: 'white',
  },
  buttonInner: {
    ...defaults.buttonInner,
    backgroundColor: 'black',
    borderColor: 'white',
    color: 'white',
  },
  infoLink: {
    ...defaults.infoLink,
    color: 'white',
  },
  infoVersion: {
    ...defaults.infoVersion,
    color: 'white',
  },
};

const stylesDefaults = StyleSheet.create(defaults);

const stylesContrast = StyleSheet.create(contrast);

export default function(useContrast: boolean = false) {
  return useContrast ? stylesContrast : stylesDefaults;
}
