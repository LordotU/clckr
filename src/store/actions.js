/**
 * @format
 * @flow
 */

type ExtractReturn<Fn> = $Call<<T>((...Iterable<any>) => T) => T, Fn>;

export const actionTypes = { SWITCH_CONTRAST: 'SWITCH_CONTRAST' };

export const switchContrast = (highContrast: boolean) => ({
  type: actionTypes.SWITCH_CONTRAST,
  payload: highContrast,
});

export type Action = ExtractReturn<typeof switchContrast>;
