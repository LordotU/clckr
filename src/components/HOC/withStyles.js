/* eslint-disable react-hooks/rules-of-hooks */

/**
  @flow
 * @format
 */

import React, { useContext } from 'react';

import { context } from '../../store';
import useStyles from '../../Styles';

export default function(Component) {
  return function withStyles(props) {
    const {
      state: { highContrast },
    } = useContext(context);

    const styles = useStyles(highContrast);

    return <Component styles={styles} {...props} />;
  };
}
