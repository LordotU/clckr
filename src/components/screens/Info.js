/**
 * @format
 * @flow
 */

import React, { useContext } from 'react';
import { Image, Linking, TouchableOpacity, Text, View } from 'react-native';

import { context } from '../../store';
import { version as appVersion } from '../../../package.json';

const BackIconDark = require('../../../assets/images/icon-back-dark.png');
const BackIconLight = require('../../../assets/images/icon-back-light.png');

const INFO_URL = 'https://oleg.levsh.in';

export default function({ navigation, styles }): React$Node {
  const {
    state: { highContrast },
  } = useContext(context);

  const goBack = () => navigation.goBack();
  const openUrl = () => Linking.openURL(INFO_URL);

  return (
    <>
      <TouchableOpacity
        accessible={true}
        accessibilityLabel="Show home"
        style={styles.infoButton}
        onPress={goBack}>
        <Image
          source={highContrast ? BackIconLight : BackIconDark}
          style={styles.infoIcon}
        />
      </TouchableOpacity>
      <View style={styles.info}>
        <TouchableOpacity
          accessible={true}
          accessibilityLabel={`Open ${INFO_URL}`}
          style={styles.outerInfoLink}
          onPress={openUrl}>
          <Text style={styles.infoLink}>{INFO_URL}</Text>
        </TouchableOpacity>
        <Text style={styles.infoVersion}>v{appVersion}</Text>
      </View>
    </>
  );
}
