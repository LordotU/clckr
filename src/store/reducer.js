/**
 * @format
 * @flow
 */

import { createContext } from 'react';
import { actionTypes } from './actions';
import type { Action } from './actions';

export const defaultState = {
  highContrast: false,
};

export const context = createContext<Object>(defaultState);

export const reducer = (
  state: typeof defaultState,
  { type, payload }: Action,
) => {
  switch (type) {
    case actionTypes.SWITCH_CONTRAST:
      return { ...state, highContrast: payload };
    default:
      return state;
  }
};
