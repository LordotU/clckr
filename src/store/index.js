/**
 * @format
 * @flow
 */

import React, { useReducer } from 'react';
import { context, defaultState, reducer } from './reducer';

const { Provider }: { Provider: Object } = context;

const StateProvider = ({ children }: { children: React$Node }) => {
  const [state, dispatch] = useReducer(reducer, defaultState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

export { context, StateProvider };
