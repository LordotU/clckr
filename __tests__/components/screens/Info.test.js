/**
 * @format
 */

import React from 'react';
import { cleanup, render } from 'react-native-testing-library';

import Info from '../../../src/components/screens/Info';
import withStyles from '../../../src/components/HOC/withStyles';
import { StateProvider } from '../../../src/store';

const InfoWithStyles = withStyles(Info);

const getInfoScreen = () => (
  <StateProvider>
    <InfoWithStyles />
  </StateProvider>
);

afterEach(cleanup);

it('renders correctly', () => {
  const snapshot = render(getInfoScreen()).toJSON();
  expect(snapshot).toMatchSnapshot();
});
