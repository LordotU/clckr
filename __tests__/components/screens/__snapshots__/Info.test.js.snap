// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    accessibilityLabel="Show home"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 8,
        "position": "absolute",
        "right": 24,
        "top": 40,
        "zIndex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/icon-back-dark.png",
        }
      }
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="Open https://oleg.levsh.in"
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "WorkSans-Thin",
            "fontSize": 52,
            "textDecorationLine": "underline",
          }
        }
      >
        https://oleg.levsh.in
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "WorkSans-Thin",
          "fontSize": 43,
        }
      }
    >
      v
      1.0.0
    </Text>
  </View>,
]
`;
